Make ProgressScope nodiscard

ddn0 opened the pull request:
It is probably a bug to not capture a ProgressScope in a local variable
as the alternative would create and immediately close a span. Use
nodiscard to enforce this policy.

roshandathathri commented on libquery/test/CMakeLists.txt:
```diff
@@ -8,6 +8,16 @@ target_link_libraries(filter-op-test galois_query)
 add_test(NAME filter-op-test COMMAND filter-op-test)
 set_tests_properties(filter-op-test PROPERTIES LABELS quick)
 
+add_executable(execution-plan-test ExecutionPlanTest.cpp)
```
Intentional?

bowu replied:
I'll remove it. Actually, I don't know how it was added.

roshandathathri replied:
Thought so.

bowu commented on libquery/src/ConstOp.cpp:
```diff
@@ -1,5 +1,7 @@
 #include "querying/ops/ConstOp.h"
 
+uint32_t galois::ConstOp::num_const_ops_ = 0;
```
Can we do this in the declaration? It's not static const.

roshandathathri replied:
Yes, it need not be const.

bowu replied:
It does't work: "Non-const static data member must be initialized out of line".

roshandathathri replied:
Oh.. my bad! Sorry.

roshandathathri reviewed (approved):
LGTM

ddn0 committed:
Make ProgressScope nodiscard

It is probably a bug to not capture a ProgressScope in a local variable
as the alternative would create and immediately close a span. Use
nodiscard to enforce this policy.

patrickkenney9801 reviewed (approved):
This is great, did not know we could do this in C++
